code/script.js

```diff
@@ -3,17 +3,156 @@
 // Step 1 - Welcome and introduction
 // Your code goes here
 alert(
-  `Welcome to our Javascript Pizzeria. Ready to Start? - Click 'OK' to begin.`
-)
+  `Welcome to our Javascript Pizzeria.🍕 Ready to Start? - Click 'OK' to begin.`
+);
+
+let userName = prompt("Write your name in the box below 👇");
+alert("👋 Hello, " + userName + "! Click 'OK' to unlock the menu!");
+console.log("Hello, " + userName + "!");
 
 // Step 2 - Food choice
 // Your code goes here
+const foodType = prompt(
+  `Choose your desired dish from the options below. Write the number of your choice and click 'OK'
+  1 - Pizza 🍕
+  2 - Pasta 🍝
+  3 - Salad 🥗
+  `
+);
+
+let selectedDish = "";
+
+if (foodType === "1") {
+  selectedDish = "Pizza 🍕";
+} else if (foodType === "2") {
+  selectedDish = "Pasta 🍝";
+} else if (foodType === "3") {
+  selectedDish = "Salad 🥗";
+} else {
+  alert(
+    "🫢 Oops! It seems like you didn't select a dish. Please try again and write the corresponding number in the box."
+  );
+}
+
+if (selectedDish !== "") {
+  alert(
+    `You've selected ${selectedDish}. Click 'OK' to explore available options.`
+  );
+}
+
+console.log("type of food", selectedDish);
 
 // Step 3 - Subtype choice
 // Your code goes here
+let selectedSubtype = "";
+let subtypeNames = {};
 
-// Step 4 - Age
-// Your code goes here
+if (foodType === "1") {
+  // Pizza
+  subtypeNames = {
+    1: "Margherita",
+    2: "Pepperoni",
+    3: "Hawaiian",
+  };
+  selectedSubtype = prompt(
+    `🍕 Choose your desired type of pizza. Write the number of your choice and click 'OK'.
+     1 - Margherita 
+     2 - Pepperoni 
+     3 - Hawaiian 
+    `
+  );
+} else if (foodType === "2") {
+  // Pasta
+  subtypeNames = {
+    1: "Fettucine",
+    2: "Pesto",
+    3: "Carbonara",
+  };
+  selectedSubtype = prompt(
+    `🍝 Choose your desired type of pasta. Write the number of your choice and click 'OK'.
+     1 - Fettucine
+     2 - Pesto
+     3 - Carbonara
+    `
+  );
+} else if (foodType === "3") {
+  // Salad
+  subtypeNames = {
+    1: "Caesar",
+    2: "Greek",
+    3: "Caprese",
+  };
+  selectedSubtype = prompt(
+    `🥗 Choose your desired type of salad. Write the number of your choice and click 'OK'.
+     1 - Caesar 
+     2 - Greek 
+     3 - Caprese 
+    `
+  );
+}
+
+if (subtypeNames[selectedSubtype]) {
+  alert(
+    `You've selected ${subtypeNames[selectedSubtype]} ${
+      foodType === "1" ? "pizza" : foodType === "2" ? "pasta" : "salad"
+    }. Excellent choice! ✨ Click 'OK' to continue.`
+  );
+} else {
+  alert(
+    `🫢 Oops! Invalid selection for ${
+      foodType === "1" ? "pizza" : foodType === "2" ? "pasta" : "salad"
+    } subtype. Please try again and write the corresponding number in the box.`
+  );
+}
```

Comment:
would be nice after this case of invalid choice to stop executing, consider this:
```
if (subtypeNames[selectedSubtype] !== undefined) {
  alert(
    `You've selected ${subtypeNames[selectedSubtype]} ${
      foodType === "1" ? "pizza" : foodType === "2" ? "pasta" : "salad"
    }. Excellent choice! ✨ Click 'OK' to continue.`
  );
} else  {
  alert(
    `🫢 Oops! Invalid selection for ${
      foodType === "1" ? "pizza" : foodType === "2" ? "pasta" : "salad"
    } subtype. Please try again and write the corresponding number in the box.`
  );
  return;
}

```